src/node_modules/ringbuffer.js

```diff
@@ -0,0 +1,25 @@
+import {EventEmitter} from 'events';
+
+export class RingBuffer extends EventEmitter {
+	constructor(opts) {
+		super();
+		this.records = [];
+		this.limit   = opts.bufferSize;
+	}
+
+	write(log) {
+		this.records.push(log);
+		if (this.records.length > this.limit) {
```

Comment:
Firing an event here is recommended, so interested parties can be aware.  Currently the ring buffer creates a new "silent failure" case where logs start getting discarded, but no flags are raised to notify anyone.    

If an event were emitted, an listener could take action to email or or page someone, etc, to let someone know that logs have started being discarded.

The silent failure itself could be a security issue, as attacker aware of the design could to work to generate enough of a volume of logging to cause incriminating logs to be silently discarded.